src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/versioning/VersionService.java

```diff
@@ -0,0 +1,76 @@
+package org.jboss.xavier.integrations.migrationanalytics.business.versioning;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.EnumerablePropertySource;
+import org.springframework.core.env.Environment;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+
+@Named
+public class VersionService {
+    @Inject
+    private Environment env;
+
+    @Getter
+    @Setter
+    private Map<String, String> properties;
+
+    @PostConstruct
+    public void init() {
+        properties = getAllProperties();
+    }
+
+    public String getPropertyWithFallbackVersion(String payloadVersion, String path) {
+        String fallbackVersion = getFallbackVersionPath(payloadVersion, path);
+
+        return properties.get("cloudforms.manifest.v" + fallbackVersion + "." + path);
+    }
+
+    public String getFallbackVersionPath(String payloadVersion, String path) {
+        ManifestPathVersion fallbackversion = properties.keySet()
+                .stream()
+                .filter(e -> e.matches("cloudforms.manifest.v[0-9_]*." + path))
+                .map(e -> e.split("\\.")[2])
+                .map(e -> e.replace("v", ""))
+                .map(this::expandVersion)
+                .map(ManifestPathVersion::new)
+                .sorted(Comparator.reverseOrder())
+                .filter(e -> isVersionEqualOrLower(payloadVersion, e))
+                .findFirst()
+                .orElse(new ManifestPathVersion("0_0_0"));
+
+        return fallbackversion.getVersion();
+    }
+
+    private boolean isVersionEqualOrLower(String payloadVersion, ManifestPathVersion e) {
+        return e.compareTo(expandVersion(payloadVersion)) <= 0;
+    }
+
+    private Map<String, String> getAllProperties() {
+        return StreamSupport.stream(((AbstractEnvironment) env).getPropertySources().spliterator(), false)
+                    .filter(ps -> ps instanceof EnumerablePropertySource)
+                    .map(ps -> ((EnumerablePropertySource) ps).getPropertyNames())
+                    .flatMap(Arrays::stream)
+                .filter(e -> e.matches("cloudforms.manifest.v[0-9_]*.*"))
+                .collect(Collectors.toMap( this::getKey, e-> env.getProperty(e)));
+    }
+
+    private String getKey(String property) {
+        String[] elements = property.split("\\.");
+        return property.replace(elements[2], "v" + expandVersion(elements[2].replace("v", "")));
```

Comment:
I've got the impression that, in the end, the `v` char is useless and just an obstacle that you had to remove and add every time (not just here: for example also `Calculator.java` in https://github.com/project-xavier/xavier-integration/pull/58/files#diff-d0deeb2ee90a27aed5b62aa3e965c5f3L23 has a `replace("v", "")` as well)
It seems like everything will work without the need of managing the `v` if we would have properties in the format `cloudforms.manifest.1_0_0.something.else=path` because there's no need for the `v` afaiu.
If it's true, please remove the `v` from the algorithm and the properties.

Reply:
I will check that

Reply:
ok